vyper/utils.py

```diff
@@ -194,6 +194,9 @@ def in_bounds(cls, type_str, value):
     "~codelen",
     "label",
     "goto",
+    "~extcode",
+    "~selfcode",
+    "~calldata",
```

Comment:
@charles-cooper These are not exactly suggested in https://github.com/vyperlang/vyper/pull/2583#pullrequestreview-847198613, but I chose these since I noticed that `msg.data` could be also used as `len(msg.data)`, so `~calldata_slice` and `~calldata_len` will be needed, which would required to be distinguished in `Expr.parse_Attribute` by looking up its parent node etc...
I thought such process would be unnecessary complication.
What do you think about this?

Reply:
> @charles-cooper These are not exactly suggested in [#2583 (review)](https://github.com/vyperlang/vyper/pull/2583#pullrequestreview-847198613), but I chose these since I noticed that `msg.data` could be also used as `len(msg.data)`, so `~calldata_slice` and `~calldata_len` will be needed, which would required to be distinguished in `Expr.parse_Attribute` by looking up its parent node etc... I thought such process would be unnecessary complication. What do you think about this?

Right I saw that! I think your reasoning is good.